client/remote.go

```diff
@@ -28,88 +26,69 @@ type DaemonConfig struct {
-	r *RemoteVPS
-}
-
-// NewSSHRunner returns a new SSHRunner
-func NewSSHRunner(r *RemoteVPS) *SSHRunner {
-	return &SSHRunner{r: r}
+// GetHost creates the user@IP string.
+func (remote *RemoteVPS) GetHost() string {
+	return remote.User + "@" + remote.IP
 }
 
-// Run runs a command remotely.
-func (runner *SSHRunner) Run(cmd string) (*bytes.Buffer, *bytes.Buffer, error) {
-	session, err := getSSHSession(runner.r.PEM, runner.r.IP, runner.r.Daemon.SSHPort, runner.r.User)
-	if err != nil {
-		return nil, nil, err
-	}
-	// Capture result.
-	var stdout, stderr bytes.Buffer
-	session.Stdout = &stdout
-	session.Stderr = &stderr
-
-	err = session.Run(cmd)
-	return &stdout, &stderr, err
+// GetIPAndPort creates the IP:Port string.
+func (remote *RemoteVPS) GetIPAndPort() string {
+	return remote.IP + ":" + remote.Daemon.Port
 }
 
 // Bootstrap configures a remote vps for continuous deployment
 // by installing docker, starting the daemon and building a
 // public-private key-pair. It outputs configuration information
 // for the user.
 func (remote *RemoteVPS) Bootstrap(runner SSHSession, name string, config *Config) error {
-	println("Bootstrapping remote " + name)
+	println("Setting up remote " + name + " at " + remote.IP)
 
 	// Generate a session for each command.
-	println("Installing docker")
-	err := remote.InstallDocker(runner)
+	println(">> Step 1/4: Installing docker...")
+	err := remote.installDocker(runner)
 	if err != nil {
 		return err
 	}
 
-	println("Starting daemon")
+	println("\n>> Step 2/4: Building deploy key...")
 	if err != nil {
 		return err
 	}
-	err = remote.DaemonUp(runner, config.Version, remote.Daemon.Port)
+	pub, err := remote.keyGen(runner)
 	if err != nil {
 		return err
 	}
 
-	println("Building deploy key\n")
+	println("\n>> Step 3/4: Starting daemon...")
 	if err != nil {
 		return err
 	}
-	pub, err := remote.KeyGen(runner)
+	err = remote.DaemonUp(runner, config.Version, remote.IP, remote.Daemon.Port)
 	if err != nil {
 		return err
 	}
 
-	println("Fetching daemon API token")
-	token, err := remote.GetDaemonAPIToken(runner, config.Version)
+	println("\n>> Step 4/4: Fetching daemon API token...")
+	token, err := remote.getDaemonAPIToken(runner, config.Version)
 	if err != nil {
 		return err
 	}
-
 	remote.Daemon.Token = token
 	err = config.Write()
 	if err != nil {
 		return err
 	}
 
-	println("Daemon running on instance")
+	println("Inertia has been set up and daemon is running on remote!\n")
+
+	println("=============================\n")
 
 	// Output deploy key to user.
-	println("GitHub Deploy Key (add here https://www.github.com/<your_repo>/settings/keys/new): ")
+	println("GitHub Deploy Key (add to https://www.github.com/<your_repo>/settings/keys/new): ")
 	println(pub.String())
 
 	// Output Webhook url to user.
-	println("GitHub WebHook URL (add here https://www.github.com/<your_repo>/settings/hooks/new): ")
+	println("GitHub WebHook URL (add to https://www.github.com/<your_repo>/settings/hooks/new): ")
```

Comment:
Might be nice to backfill the repo name here (if that’s an option now)

Reply:
#123